Source/Core/Core/State.cpp

```diff
@@ -94,7 +94,7 @@ static size_t s_state_writes_in_queue;
 static std::condition_variable s_state_write_queue_is_empty;
 
 // Don't forget to increase this after doing changes on the savestate system
-constexpr u32 STATE_VERSION = 149;  // Last changed in PR 10781
+constexpr u32 STATE_VERSION = 150;  // Last changed in PR 11124 on October 6, 2022
```

Comment:
Specifying the date isin't particularly necessary given that the date is contained in the commit and the pr and so on, fwiw. 

Also, wrong PR number btw.

Reply:
Yeah, I kind of got sniped there by Pokechu's PR, lol!

I just added the date because it was annoying me earlier when I wanted to figure out how long it had been since the savestate version was last modified by just looking at it in Visual Studio.

Reply:
Either way, I fixed it.

Reply:
I don't see why knowing the date would be useful. Also, you can enable the Blame view in Visual Studio if you want to see the date the line was last changed.

Reply:
From the perspective of TASers trying to look into the Savestate code (and who don't have very much experience looking through the Dolphin code/using visual studio), it seems like it would be a helpful add-on. However, if this PR won't get merged with the date in it, then I can remove it.

Reply:
I think the issue with the date is that it reflects when the PR was *made*, not when it was *merged*, making it a bit less useful.

I would say the PR number itself is not that useful too, but it does have a benefit: two PRs that both change the version from 149 to 150 would not create a merge conflict after the first PR is merged, but _with_ the PR number you _will_ get a merge conflict, which is good because the second PR would need to instead change the number from 150 to 151. Including the PR number thus forces you to update the state version if another PR changed it and was merged.

You can use the blame view in GitHub's web UI too: https://github.com/dolphin-emu/dolphin/blame/master/Source/Core/Core/State.cpp#L96

Reply:
In that case, I'd better finish this PR up and get it merged tonight so that the date will be accurate in any event!

Reply:
Updated the PR to include initializer + shutdown method for MemoryInterface, and fixed the savestate version number (also removed the date).